Stable-Branch-Strategy.md

```diff
@@ -0,0 +1,95 @@
+Branch and release maintenance for Kata Containers project
+
+# Overview
+
+As detailed in the release documentation, the Kata Containers project makes use of
+semantic versioning.  A release version is described by MAJOR.MINOR.PATCH.  Early in
+the project, we will introduce many new features which will require a minor version change
+as well as bug fixes.  To facilitate a stable user environment, Kata will begin to provide
+stable branch based releases as well as a master branch release.
+
+## Branch Management
+Kata Containers will maintain two stable release branches in addition to the master branch.
+Once a new MAJOR or MINOR release is created off of master, a new stable branch will be created for
+the prior MAJOR or MINOR release and the older stable branch will no longer be maintained.
+
+A couple of examples are provided to help clarify this process.
+
+### New bug fix introduced
+Consider the following initial state:
+Master 1.3.rc0
+Stable 1.2.0
+Stable 1.1.2
+ 
+A bug fix is submitted against the runtime which does not introduce new inter-component dependencies.
+This fix will be applied to both the master and stable branch, and the new state will be:
+```
+Master: 1.3.rc1 
+Stable: 1.2.1
+Stable: 1.1.3
+```
+
+### New release made feature or change which  adds new inter-component dependency
+Consider the following initial state:
+```
+Master: 1.3.rc1
+Stable: 1.2.1
+Stable: 1.1.3
+```
+A new feature is introduced which adds a new inter-component dependency. The new branch state will be:
+```
+Master: 1.3.0
+Stable: 1.3.0
+Stable: 1.2.2
+```
+ 
+Note, the 1.1.3 branch will still exist, but under current plans it would not be maintained further. 
+   
+## Backporting Process 
+
+Development will occur against the master branch and applicable code commits should also be submitted
+against the stable branch. Some guidelines:
+-Only bug and security fixes which do not introduce inter-component dependencies are
```

Comment:
This would be clearer as a bullet list or table maybe.

Also, to support this process, I wonder if we should start labelling PRs with `bugfix` or `feature` to make this backporting easier? Ideally, PR authors would add the appropriate label, but otherwise the [Review team](https://github.com/kata-containers/community/wiki/Review-Team-Rota) could add in any missing ones.

We could also mandate such labels by adding a CI check to ease the pain of backporting further since it should be trivial for anyone to identify whether a PR is a bug fix or a new feature. If it's both, "new feature" would "trump" (override) bug-fix but we could further require that PRs do not contain combinations of commits - PRs must be either "pure bug fix" or "pure new feature" [1].

/cc @grahamwhaley, @bergwolf.

---

[1] - Github users who created such PRs could then be referred to as "Pure virtual constructors". Sorry... :smile: